File: apzkr-pzpi-21-4-bondarenko-taras/Task4-MobileClient/CargoTruckMobileClient/app/src/main/java/com/example/cargotruckmobile/Network/UserService.kt
package com.example.cargotruckmobile.Network

import com.example.cargotruckmobile.Models.Credentials
import com.example.cargotruckmobile.Models.LoginResponse
import com.example.cargotruckmobile.Models.User
import com.example.cargotruckmobile.Models.UserCreate
import com.example.cargotruckmobile.Models.UserDto
import com.example.cargotruckmobile.Models.UserListResponse
import com.example.cargotruckmobile.Models.UserUpdate
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface UserService {
    @POST("users/register")
    suspend fun registerUser(@Body user: UserCreate): retrofit2.Response<Void>

    @POST("users/login")
    suspend fun loginUser(@Body credentials: Credentials): retrofit2.Response<LoginResponse>


    @GET("users/{id}")
    suspend fun getUser(@Path("id") id: String): User

    @GET("users/Number/{numberPhone}")
    suspend fun getUserByNumberPhone(@Path("numberPhone") numberPhone: String): UserDto

    @PUT("users")
    suspend fun updateUser(@Body user: UserUpdate): retrofit2.Response<Void>

    @PUT("users/ban/{id}")
    suspend fun banUser(@Path("id") id: String): retrofit2.Response<Void>

    @PUT("users/unban/{id}")
    suspend fun unBanUser(@Path("id") id: String): retrofit2.Response<Void>

    @GET("users?pageNumber=1&pageSize=100")
    suspend fun getUser(): UserListResponse
}

File: apzkr-pzpi-21-4-bondarenko-taras/Task3-WebClient/CargoTruckWebClient/src/Api/UserService.js
import { ApiClient } from './ApiClient';

export const UserService = {
  registerUser: async (user) => {
    const response = await ApiClient.post('users/register', user);
    return response;
  },
  loginUser: async (credentials) => {
    const response = await ApiClient.post('users/login', credentials);
    return response;
  },
  getUser: async (id) => {
    const response = await ApiClient.get(`users/${id}`);
    return response.data;
  },
  getUserByNumberPhone: async (numberPhone) => {
    const response = await ApiClient.get(`users/Number/${numberPhone}`);
    return response.data;
  },
  updateUser: async (user) => {
    const response = await ApiClient.put('users', user);
    return response;
  },
  getPageUser: async () => {
    const response = await ApiClient.get('users?pageNumber=1&pageSize=100');
    return response.data.items;
  },
  deleteUser: async (id) => {
    const response = await ApiClient.delete(`users/delete/${id}`);
    return response;
  },
  banUser: async (id) => {
    const response = await ApiClient.put(`users/ban/${id}`);
    return response;
  },
  unBanUser: async (id) => {
    const response = await ApiClient.put(`users/unban/${id}`);
    return response;
  },
};
